R/generic-spec.R

```diff
@@ -28,7 +28,8 @@ as_S3_generic <- function(x) {
     }
   }
 
-  stop("`generic` is a function, but not an S3 generic function", call. = FALSE)
+  stop("`generic` is a function, but not an S3 generic function: ",
+       deparse1(x, width.cutoff = 100L), call. = FALSE)
```

Comment:
I think displaying the arguments, without the body, would be sufficient. (Like `args(mean)` but without the `NULL`).

Reply:
`function (x, ...)` doesn't seem like a very useful breadcrumb, especially if we can easily include the head of the first expression in the function body.

Reply:
I guess I'm worried about this spitting out a very large function body, making it hard to see the actual error message.

Reply:
> `width.cutoff = 100L` 

maybe we truncate this more/better?

Reply:
I think that just determines the line length?

Reply:
TIL! (I had thought `width.cutoff` capped `nchar()` of the returned string)

Maybe something like:

```
deparse_trunc <- function(x, width, collapse = "\n") {
  x <- deparse1(x, collapse)
  if (nchar(x)> width) 
     x <- sprintf("%s....", substr(x, 0, width-4))
  x
}
```
which gives:
```r
> cat(deparse_trunc(plot.default, 80))
function (x, y = NULL, type = "p", xlim = NULL, ylim = NULL, log = "", main ....
```

Reply:
Yeah, that sounds good to me.